Gems/Atom/Utils/Code/Source/MaterialUtils.cpp

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <Atom/Utils/MaterialUtils.h>
+
+namespace AZ::Render
+{
+    
+    Matrix3x3 CreateUvTransformMatrix(const UvTransformDescriptor& desc, AZStd::span<const TransformType> transformOrder)
+    {
+        float translateX = desc.m_translateX;
+        float translateY = desc.m_translateY;
+
+        if (desc.m_scaleX != 0.0f)
+        {
+            translateX *= (1.0f / desc.m_scaleX);
+        }
+
+        if (desc.m_scaleY != 0.0f)
+        {
+            translateY *= (1.0f / desc.m_scaleY);
+        }
```

Comment:
Should translateX and translateY get set to 0 if scale is 0? i.e.
```
translateX *= (desc.m_scaleX != 0.0f) ? (1.0f / desc.m_scaleX) : 0.0f;
translateY *= (desc.m_scaleY != 0.0f) ? (1.0f / desc.m_scaleY) : 0.0f;
```

Reply:
I don't think translation really has any meaning if the scale is 0. This code was also just moved from the functor, so I want to be careful not to change any functionality